test/controllers/benefits/CompanyBenefitControllerSpec.scala

```diff
@@ -176,10 +186,16 @@ class CompanyBenefitControllerSpec
     }
 
     "redirect to the appropriate IFORM update page" when {
-      "the form has the value yesIGetThisBenefit" in {
+      "the form has the value yesIGetThisBenefit and EndCompanyBenefit_BenefitTypeKey is cached" in {
 
         val SUT = createSUT
 
+        val benefitType = "Telephone"
```

Comment:
To be explicit about it being a benefit you could Telephone type in `TaxComponentType`